src/main/java/udpapi2/UdpApi.java

```diff
@@ -0,0 +1,341 @@
+        executorService.schedule(new Send<>(this, command, aniDbIp, aniDbPort), delay.toMillis(), TimeUnit.MILLISECONDS);
+        isSendScheduled = true;
+    }
+
+    private Duration getNextSendDelay() {
+        if (shouldWaitLong) {
+            return UdpApiConfiguration.LONG_WAIT_TIME;
+        }
+        if (lastSentDate == null) {
+            return Duration.ZERO;
+        }
+        val nextSend = UdpApiConfiguration.COMMAND_INTERVAL_MS.minus(Duration.ofMillis(new Date().getTime() - lastSentDate.getTime()));
+        if (nextSend.isNegative()) {
+            return Duration.ZERO;
+        }
+        return nextSend;
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (socket != null) {
+            socket.close();
+            socket = null;
+        }
+        isInitialized = false;
+    }
+
+    @Override
+    public void addReply(Reply reply) {
+        val query = queries.get(reply.getFullTag());
+        if (query == null) {
+            logger.warning(STR."Reply without corresponding query \{reply.toString()}");
+            return;
+        }
+        query.setReply(reply);
+        handleQueryReply(query);
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void handleQueryReply(Query query) {
+        if (replyStatusCallbacks.containsKey(query.getReply().getReplyStatus())) {
+            replyStatusCallbacks.get(query.getReply().getReplyStatus()).invoke(query.getReply().getReplyStatus());
+        }
+        if (!query.success()) {
+            logger.warning(STR."Query failed: \{query.toString()}");
+            handleQueryError(query);
+            return;
+        }
+        queries.remove(query.getFullTag());
+
+        val command = query.getCommand();
+
+        if (command instanceof LoginCommand) {
+            switch (query.getReply().getReplyStatus()) {
+                case LOGIN_ACCEPTED, LOGIN_ACCEPTED_NEW_VERSION -> {
+                    session = query.getReply().getResponseData().getFirst();
+                    logger.info(STR."Logged in with session \{session}");
+                    loginStatus = LoginStatus.LOGGED_IN;
+                }
+            }
+        } else if (command instanceof LogoutCommand) {
+            switch (query.getReply().getReplyStatus()) {
+                case LOGGED_OUT, NOT_LOGGED_IN -> {
+                    session = null;
+                    logger.info("Logged out");
+                    loginStatus = LoginStatus.LOGGED_OUT;
+                    if (shutdown) {
+                        shutdown();
+                        return;
+                    }
+                }
+            }
+        }
+        if (commandCallbacks.containsKey(command.getClass())) {
+            //noinspection unchecked
+            commandCallbacks.get(command.getClass()).invoke(query);
+        } else {
+            logger.warning(STR."Unhandled query reply: \{query.toString()}");
```

Comment:
Not sure if you expect it to normally log this warning, but it is happening for me :)
```
2024-07-21 10:38:26 FINE processing.Mod_EpProcessing processEps Processing file Z:\xxx\anidbclitest1\A Silent Voice [anidb-10937]\A Silent Voice [BlurayDesuYo][BluRay][1920x1080][HEVC][7273A66D].mkv with Id 0
2024-07-21 10:39:39 FINE processing.Mod_EpProcessing onHashComputed File Z:\xxx\anidbclitest1\A Silent Voice [anidb-10937]\A Silent Voice [BlurayDesuYo][BluRay][1920x1080][HEVC][7273A66D].mkv with Id 0 has been hashed
2024-07-21 10:39:39 INFO udpapi2.UdpApi scheduleCommand Scheduling command Command(action=AUTH, identifier=auth, tag=null, needsLogin=false, parameters={client=aniaddcli, clientver=4, comp=1, enc=UTF8, nat=1, pass=****, protover=3, user=****}, queryId=1) in 0 ms
2024-07-21 10:39:39 FINE udpapi2.Send run Send command auth
2024-07-21 10:39:39 FINE processing.Mod_EpProcessing onHashComputed Requested Data for file with Id 0: SendFile: true, SendML: false
2024-07-21 10:39:39 FINE processing.Mod_EpProcessing processEps Processing file Z:\xxx\anidbclitest1\A Silent Voice [anidb-10937]\A Silent Voice [Okay-Subs][BluRay][1920x1036][HEVC][6457E1C3].mkv with Id 1
2024-07-21 10:39:39 FINE udpapi2.ParseReply parseReply Reply: auth-1 200 RKbYk xxxx.xxx.xxx.xxx:3333 LOGIN ACCEPTED \n
2024-07-21 10:39:39 INFO udpapi2.UdpApi handleQueryReply Logged in with session RKbYk
2024-07-21 10:39:39 WARNING udpapi2.UdpApi handleQueryReply Unhandled query reply: Query(command=Command(action=AUTH, identifier=auth, tag=null, needsLogin=false, parameters={client=aniaddcli, clientver=4, comp=1, enc=UTF8, nat=1, pass=****, protover=3, user=****}, queryId=1), reply=Reply(fullMessage=auth-1 200 RKbYk xxx.xxx.xxx.xxx:3333 LOGIN ACCEPTED
, fullTag=auth-1, replyStatus=LOGIN_ACCEPTED, queryId=null, responseData=[RKbYk, xxx.xxx.xxx.xxx:3333, LOGIN, ACCEPTED
]), sentAt=Sun Jul 21 10:39:39 CDT 2024, receivedAt=null, retries=0)
```

Reply:
yeah currently that's expected, i'll have to change that to make sure only actually undhandled replies trigger this. (login is handled by the API itself, all other commands must (should) be handled by the consumer of the api)

Reply:
i've added a todo for now, will fix this week